### ISSUE TITLE
F-droid repository

### ISSUE THREAD

**Issue by im7mortal** [amnezia-vpn/amnezia-client#263]
Amnezia VPN is an excellent application. However, its current distribution on privacy-oriented platforms like Linux and degoogled Android leaves much to be desired. While I understand that Amnezia VPN aims for easy setup for a wide audience, I believe it's essential to stay closely connected with the privacy community for apps like this, to garner the best feedback and ideas.

As I couldn't locate an APK on the repo, I forked it and set up actions to create an unsigned APK. Then, using F-Droid server tools, I signed and deployed it.

**[There is my AmneziaVPN f-droid repository.](https://d2kntb75wqejvh.cloudfront.net/fdroid/repo/)**

The process is quite simple to automate, and I plan to do so.

There are two ways to distribute apps with F-Droid:

1. Own Repo (current setup): This approach allows apps to be signed with F-Droid or built from scratch.

2. Main F-Droid Repo: This method requires setting up metadata on GitLab, containing the build recipe and source code link. The app must be FOSS, which is not the primary focus of the project. The advantage is that the app will be available on the main repo, but it's not critically important, as those who need it can download it from the repository.

**Comment by marek22k** [comment]
If it's not in the official F-Droid repo yet, maybe it could be added to IzzyOnDroid? @IzzySoft

**Comment by IzzySoft** [comment]
Sorry @marek22k – but the APK size is more than twice the per-app size limit, so I'm afraid at least currently that's not possible.

**Comment by marek22k** [comment]
@IzzySoft Thanks for the reply! That is understandable.

There is also already a request to include the app in the main F-Droid repo: https://gitlab.com/fdroid/rfp/-/issues/2374

**Comment by OmlineEditor** [comment]
In the settings of this application, there is an option to create an Amnesia tunnel. try to test this application if you only need a tunnel, without additional settings.

https://f-droid.org/en/packages/com.zaneschepke.wireguardautotunnel/

**Comment by Korb** [comment]
_@OmlineEditor, It's a pity that WG Tunnel does not support adding configurations by pasting a key from the clipboard._

**Comment by marek22k** [comment]
By key do you mean a `vpn://`-URL? Maybe just create an FR in the WG Tunnel repo? Or alternatively convert the `vpn://` file into a .conf file [manually](https://codeberg.org/mark22k/get_amnezia_free)?

**Comment by Korb** [comment]
> the APK size is more than twice the per-app size limit, so I'm afraid at least currently that's not possible.

Is there a limit on APK size in F-Droid? Help me find it in the official documentation, please.

**Comment by marek22k** [comment]
As far as I know, the official F-Droid repo has no limit. Only IzzyOnDroid, who runs his own F-Droid repo, has one.

**Comment by IzzySoft** [comment]
Correct.

**Comment by vsolontsov-ll** [comment]
> [@IzzySoft](https://github.com/IzzySoft) Thanks for the reply! That is understandable.
> 
> There is also already a request to include the app in the main F-Droid repo: https://gitlab.com/fdroid/rfp/-/issues/2374

As far as I can get, it's pending on "The original app author has been notified (and does not oppose the inclusion)."
There're 2 comments from @marek22k liking to this issue.  So formally this requirement should be marked as satisfied I think. Yet, no progress.

I'm wondering how this can be pushed?

**Comment by IzzySoft** [comment]
AFAIK only by opening your own merge request at fdroiddata…

**Comment by ismay-tue** [comment]
So should we open a new issue at the gitlab repo? It seems like the current one isn't going anywhere. It's been years, and it seems to be stuck on nothing really..

**Comment by OmlineEditor** [comment]
> So should we open a new issue at the gitlab repo? It seems like the current one isn't going anywhere. It's been years, and it seems to be stuck on nothing really..

If you know how to do this so that things go smoothly and we get results, please do it.

**Comment by ismay-tue** [comment]
Well I suppose I'd start by asking what it's stuck on. But in the meantime I've found https://wgtunnel.com/, and that does the job just fine. So the immediate need to get this resolved isn't there for me anymore.
